rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -920,6 +934,32 @@ RMW_Connext_Publisher::assert_liveliness()
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+RMW_Connext_Publisher::wait_for_all_acked(rmw_time_t wait_timeout)
+{
+  DDS_Duration_t timeout = DDS_DURATION_INFINITE;
+
+  if (!rmw_time_equal(wait_timeout, RMW_DURATION_INFINITE)) {
+    rmw_connextdds_duration_from_ros_time(&timeout, &wait_timeout);
+  }
```

Comment:
shouldn't the rmw_connextdds_duration_from_ros_time() function handle the infinity case already?

Reply:
The definition `DDS_Duration_t` is http://community.rti.com/rti-doc/510/ndds/doc/html/api_cpp/structDDS__Duration__t.html.
DDS_Long and DDS_UnsignedLong are 32-bit integer value.   
I think `rmw_connextdds_duration_from_ros_time()` cannot  covert infinite duration. So if the duration is infinite, directly use `DDS_DURATION_INFINITE`.

Reply:
@ivanpauno  friendly ping.

Reply:
> I think rmw_connextdds_duration_from_ros_time() cannot correctly covert infinite duration (64-bit to 32-bit). So if the duration is infinite, directly use DDS_DURATION_INFINITE.

Sure, I mean that it should be handled inside `rmw_connextdds_duration_from_ros_time()` and not here.

Reply:
See https://github.com/ros2/rmw_connextdds/pull/20#discussion_r642479233 and https://github.com/ros2/rmw_connextdds/pull/20#discussion_r642480389.

Reply:
Get it. Thanks.